src/main/java/gift/dto/ProductDTO.java

```diff
@@ -35,7 +35,6 @@ public class ProductDTO {
     private CategoryName categoryName;
 
     @NotNull(message = "옵션은 필수 입력 항목입니다.")
-
     private List<OptionDTO> options = new ArrayList<>();
 
     public ProductDTO() {}
```

Comment:
https://github.com/kakao-tech-campus-2nd-step2/spring-gift-enhancement/pull/401#discussion_r1685678871
이전 리뷰에 대해 생각을 정리해보았습니다. 

기본생성자를 왜 사용하는지 구글링 해보았습니다.

저와 같은 이유로 ObjectMapper를 위한 기본생성자를 사용할 것인지 에 관한 이슈들을 찾아볼수 있었습니다.
결론 부터 말씀드리자면, 
`기본 생성자는 필요하고. 빈 객체가 생성되는 문제는 private로 막는 게 최선이라 생각합니다.`
Why? -> `Reflection은 접근 제어자와 상관 없기 때문입니다.`

미션을 진행하면서 POST요청과 같은 사용자 요청 처리가 발생합니다 

요청을 하는 DTO에서 기본생성자가 필요한 이유는 다음과 같습니다.

1. 제가 작성한 코드에선 사용자로부터 요청을 받을 때 사용자가 입력한 값을 DTO로 매핑 시킬때 역직렬화를 수행합니다. 이 과정을 ObjectMapper가 수행합니다.
2. ObjectMapper는 `@RequestBody`를 통해 받은 DTO를 기본생성자를 통해서 역직렬화 수행하게 됩니다.
 이때, Reflction이 이용됩니다. 이는 기본 생성자를 통해 수행됩니다. 
4. Reflection은 접근 제어자와 상관 없이 클래스 객체를 동적으로 생성하는 JAVA API로 

> Almost all frameworks require a default(no-argument) constructor in your class because these frameworks use reflection to create objects by invoking the default constructor
> 

무조건 기본 생성자가 필요합니다.(기본생성자로 객체를 생성하기에 )

Q. 왜 ? 기본생성자로 생성하나

→ RE : Reflection은 로딩이 완료된 클래스에서 또 다른 클래스를 동적으로 로딩하여 생성자,멤버 필드, 멤버 메서드 등을 사용할 수 있도록 한다. 

java Reflction이 가져올 수 없는 정보 중 하나가 바로 생성자의 인자 정보들이기에 기본생성자 없이 파라미터가 있는 생성자만 존재하면 java Reflction이 객체를 생성할수 없게 됩니다

Reply:
정말 정말 잘 알아보셨습니다!
reflection 이라는 기술, ObjectMapper 의 동작 방식, 그리고 `@RequestBody` 가 만들어지는 과정에 대해 조금 더 잘 알게 되었네요 👏 👏 👏 👏 

이제 이 내용에서 나오는 다른 키워드를 따라서 추가 학습을 해볼 수도 있겠어요.
예를 들어 접근제한자는 public, private 외에 어떤 것들이 있고, 그들의 특징은 무엇인지 같은 내용이요.

그리고 지금 저희 기술에는 no args constructor 를 요구하는 또다른 기술이 있는데요,
그 기술은 왜 no args constructor 가 필요한지도 알아볼 수 있구요. (+ 접근제한자는 무엇까지 허용되는지)